bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/handler/NikoHomeControlAccessHandler.java

```diff
@@ -0,0 +1,300 @@
+
+        Bridge bridge = getBridge();
+        if ((bridge != null) && ThingStatus.ONLINE.equals(bridge.getStatus())) {
+            // We need to do this in a separate thread because we may have to wait for the
+            // communication to become active
+            scheduler.submit(this::startCommunication);
+        }
+    }
+
+    @Override
+    synchronized void startCommunication() {
+        NikoHomeControlCommunication nhcComm = getCommunication(getBridgeHandler());
+
+        if (nhcComm == null) {
+            return;
+        }
+
+        if (!nhcComm.communicationActive()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.communication-error");
+            return;
+        }
+
+        NhcAccess nhcAccess = nhcComm.getAccessDevices().get(deviceId);
+        if (nhcAccess == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.deviceId");
+            return;
+        }
+
+        nhcAccess.setEventHandler(this);
+
+        updateProperties(nhcAccess);
+
+        String location = nhcAccess.getLocation();
+        if (thing.getLocation() == null) {
+            thing.setLocation(location);
+        }
+
+        this.nhcAccess = nhcAccess;
+
+        initialized = true;
+        deviceInitialized();
+    }
+
+    @Override
+    void refresh() {
+        NhcAccess access = nhcAccess;
+        if (access != null) {
+            accessBellEvent(access.getBellState());
+            accessDoorLockEvent(access.getDoorLockState());
+            if (access.getType().equals(AccessType.RINGANDCOMEIN)) {
+                accessRingAndComeInEvent(access.getRingAndComeInState());
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        NikoHomeControlCommunication nhcComm = getCommunication(getBridgeHandler());
+        if (nhcComm != null) {
+            NhcAccess access = nhcComm.getAccessDevices().get(deviceId);
+            if (access != null) {
+                access.unsetEventHandler();
+            }
+        }
+        nhcAccess = null;
+        super.dispose();
+    }
+
+    private void updateProperties(NhcAccess nhcAccess) {
+        Map<String, String> properties = new HashMap<>();
+
+        if (nhcAccess instanceof NhcAccess2) {
+            NhcAccess2 access = (NhcAccess2) nhcAccess;
+            properties.put(PROPERTY_DEVICE_TYPE, access.getDeviceType());
+            properties.put(PROPERTY_DEVICE_TECHNOLOGY, access.getDeviceTechnology());
+            properties.put(PROPERTY_DEVICE_MODEL, access.getDeviceModel());
+        }
+
+        String buttonId = nhcAccess.getButtonId();
+        if (buttonId != null) {
+            properties.put("buttonId", buttonId);
+        }
+
+        if (nhcAccess.supportsVideoStream()) {
+            properties.put("username", "admin");
+            properties.put("password", "123qwe");
```

Comment:
The binding does not in itself support using the video stream from the video door device, but it could be configured using the IP Camera binding. To be able to configure, the username, password and URL's would be required. Putting this in properties is just a convenience if one wants to configure this, but not strictly required for the functioning of the binding. The video device uses a fixed password.

Reply:
Maybe rather an information to put in the documentation ?

Reply:
I would rather not write it out in the documentation, especially the password.

Reply:
Ok. If you think there is fine to have this as properties, I close my comment.

Reply:
I think it is fine.